packages/flutter/test/widgets/framework_test.dart

```diff
@@ -0,0 +1,80 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter/widgets.dart';
+
+void main() {
+  testWidgets('UniqueKey control test', (WidgetTester tester) async {
+    Key key = new UniqueKey();
+    expect(key.toString(), hasOneLineDescription);
+  });
+
+  testWidgets('ObjectKey control test', (WidgetTester tester) async {
+    Object a = new Object();
+    Object b = new Object();
+    Key keyA = new ObjectKey(a);
+    Key keyA2 = new ObjectKey(a);
+    Key keyB = new ObjectKey(b);
+
+    expect(keyA.toString(), hasOneLineDescription);
+    expect(keyA, equals(keyA2));
+    expect(keyA.hashCode, equals(keyA2.hashCode));
+    expect(keyA, isNot(equals(keyB)));
+  });
+
+  testWidgets('GlobalKey duplication', (WidgetTester tester) async {
```

Comment:
one thing we should test one day is global key duplication across different build phases in the same frame, via layout builders.